src/components/dialog/Dialog.js

```diff
@@ -7,15 +7,25 @@ import theme from './theme.css';
 
 import { Button, ButtonGroup, DialogBase, Heading2, Heading3, Link } from '../../index';
 import { COLORS } from '../../constants';
+import { IconDragMediumFilled } from '@teamleader/ui-icons';
 
 class Dialog extends PureComponent {
   bodyRef = createRef();
+  dragHandleRef = createRef();
 
   getHeader = () => {
-    const { headerColor, headerIcon, headingLevel, onCloseClick, title } = this.props;
+    const { draggable, headerColor, headerIcon, headingLevel, onCloseClick, title } = this.props;
+
+    const icon = draggable ? (
+      <div className={theme['drag-icon']} ref={this.dragHandleRef}>
+        <IconDragMediumFilled />
+      </div>
+    ) : (
+      headerIcon
+    );
 
     return (
-      <DialogBase.Header color={headerColor} icon={headerIcon} onClose={onCloseClick}>
+      <DialogBase.Header color={headerColor} icon={icon} onClose={onCloseClick}>
```

Comment:
True, but we should probably not use expose `DialogBase` anymore in the future all together and instead adapt our `Dialog` component to our needs, the `DialogBase` component is used as an escape hatch to prevent touching the main component